plugins/modules/foreman_templates_import.py

```diff
@@ -0,0 +1,185 @@
+    required: false
+    type: str
+  filter:
+    description:
+      - Sync only templates with name matching this regular expression, after I(prefix) was applied.
+      - Case-insensitive, snippets are not filtered.
+    required: false
+    type: str
+  negate:
+    description: Negate the prefix.
+    required: false
+    type: bool
+  dirname:
+    description: The directory within Git repo containing the templates.
+    required: false
+    type: str
+  locations:
+    description: REPLACE locations with given list.
+    required: false
+    type: list
+    elements: str
+  organizations:
+    description: REPLACE organizations with given list.
+    required: false
+    type: list
+    elements: str
+extends_documentation_fragment:
+  - foreman
+'''
+
+EXAMPLES = '''
+- name: Sync templates from git repo
+  foreman_templates_import:
+    repo: https://github.com/theforeman/community-templates.git
+    branch: 1.24-stable
+    associate: new
+    server_url: "https://foreman.example.com"
+    username: "admin"
+    password: "changeme"
+'''
+
+RETURN = ''' # '''
+
+from ansible.module_utils.foreman_helper import ForemanAnsibleModule
+
+
+def main():
+    module = ForemanAnsibleModule(
+        foreman_spec=dict(
+            location=dict(type='entity', flat_name='location_id'),
+            organization=dict(type='entity', flat_name='organization_id'),
+            associate=dict(choices=['always', 'new', 'never']),
+            prefix=dict(),
+            branch=dict(),
+            repo=dict(),
+            filter=dict(),
+            dirname=dict(),
+            verbose=dict(type='bool'),
+            force=dict(type='bool'),
+            lock=dict(type='bool'),
+            negate=dict(type='bool'),
+            locations=dict(type='entity_list', flat_name='location_ids'),
+            organizations=dict(type='entity_list', flat_name='organization_ids'),
+        ),
+        supports_check_mode=False,
+    )
+
+    module_params = module.clean_params()
+    module.connect()
+
+    if 'template' in module.foremanapi.resources or 'templates' in module.foremanapi.resources:
+        resource_name = 'template'
+    else:
+        raise Exception('The server does not seem to have the foreman_templates plugin installed.')
+
+    # Build a list of all existing templates of all supported types to check if we are adding any new
+    all_templates = []
+
+    template_types = ['provisioning_templates', 'report_templates', 'ptables']
+    if 'job_templates' in module.foremanapi.resources:
+        template_types.append('job_templates')
+
+    for template_type in template_types:
+        resources = module.list_resource(template_type)
+        for resource in resources:
+            all_templates.append([resource['name'], resource['id']])
+
+    result = module.resource_action(resource_name, 'import', record_change=False, params=module_params)
+    msg_templates = result['message'].pop('templates', [])
```

Comment:
I had a hard time to understand the purpose of this variable. Maybe call it something with 'report'.

Reply:
Building a list of templates that are present before import, so we can tell if any are new (because the plugin does not care it just says `imported=true` for every template that either changed or got added. IIRC, the diff is not helping in case of new)

all_existing_templates / all_present_templates / template_report / templates_before_import ?

Reply:
I think `template_import` would do.